eng/scripts/automation/GenerateAndBuildLib.ps1

```diff
@@ -284,65 +332,299 @@ function Invoke-Generate() {
         [string]$sdkfolder= ""
     )
     $sdkfolder = $sdkfolder -replace "\\", "/"
-    Push-Location $sdkfolder/src
-    dotnet build /t:GenerateCode
+    dotnet build /t:GenerateCode  $sdkfolder/src
     if ( !$? ) {
         Write-Error "Failed to generate sdk."
-        Pop-Location
         exit 1
     }
-    Pop-Location
 }
 
 function Invoke-Build() {
     param(
         [string]$sdkfolder= ""
     )
     $sdkfolder = $sdkfolder -replace "\\", "/"
-    Push-Location $sdkfolder
-    dotnet build
+    dotnet build $sdkfolder
     if ( !$? ) {
         Write-Error "Failed to build sdk. exit code: $?"
-        Pop-Location
         exit 1
     }
-    Pop-Location
 }
 
 function Invoke-Pack() {
     param(
         [string]$sdkfolder= ""
     )
     $sdkfolder = $sdkfolder -replace "\\", "/"
-    Push-Location $sdkfolder
-    dotnet pack
+    dotnet pack $sdkfolder /p:RunApiCompat=$false
     if ( !$? ) {
         Write-Error "Failed to build sdk package. exit code: $?"
-        Pop-Location
         exit 1
     }
-    Pop-Location
 }
 function Get-ResourceProviderFromReadme($readmeFile) {
-    $readmeFileRegex = "(?<specName>.*)/(?<serviceType>.*)/readme.md"
-    $readmeFileRegexWithSpec = "specification/(?<specName>.*)/(?<serviceType>.*)/readme.md"
-    try
-    {
-        if ($readmeFile -match $readmeFileRegexWithSpec)
-        {
-            return $matches["specName"], $matches["serviceType"]
+    $readmeFile = $readmeFile -replace "\\", "/"
+    $pathArray = $readmeFile.Split("/");
+
+    if ( $pathArray.Count -lt 3) {
+        Write-Error "Error: invalid readme file path."
```

Comment:
Why is it invalid? Would the user reading this error message know what they did wrong?

Reply:
Add the more description about the error: A valid readme file path should contain specName and serviceType and be of the form <specName>/<serviceType>/readme.md, e.g. specification/deviceupdate/data-plane/readme.md